php-classes/Emergence/Connectors/IdentityConsumerTrait.php

```diff
@@ -34,6 +34,14 @@ public static function beforeAuthenticate(IPerson $Person)
         return true;
     }
 
+    public static function getSAMLNameId(IPerson $Person)
+    {
+        return [
+            'Format' => 'urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress',
+            'Value' => $Person->Username.'@'.SAML2::$issuer
```

Comment:
@nbey an `IPerson` isn't guaranteed to have a Username, this method should handle that somehow, at least by throwing an exception or maybe by returning `null` to indicate that a nameId is not available